mava/systems/ippo/networks.py

```diff
@@ -198,34 +201,85 @@ def make_discrete_networks(
         critic_layer_sizes: sizes of hidden layers for the critic network
         observation_network: optional network for processing observations.
             Defaults to utils.batch_concat.
+        activate_mlp_final: Activate final layer of intermediary MLP
+        orthogonal_initialisation: Whether network weights should be
+            initialised orthogonally.
 
     Returns:
         PPONetworks class
     """
 
     num_actions = environment_spec.actions.num_values
 
-    def policy_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
-        policy_network = hk.Sequential(
-            [
-                observation_network,
-                hk.nets.MLP(policy_layer_sizes, activation=jax.nn.relu),
-                networks_lib.CategoricalHead(
-                    num_values=num_actions, dtype=environment_spec.actions.dtype
-                ),
-            ]
-        )
-        return policy_network(inputs)
-
-    def critic_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
-        critic_network = hk.Sequential(
-            [
-                observation_network,
-                hk.nets.MLP(critic_layer_sizes, activation=jax.nn.relu),
-                ValueHead(),
-            ]
-        )
-        return critic_network(inputs)
+    if orthogonal_initialisation:
+
+        def policy_fn(inputs: jnp.ndarray) -> networks_lib.FeedForwardNetwork:
+            policy_network = hk.Sequential(
+                [
+                    observation_network,
+                    hk.nets.MLP(
+                        policy_layer_sizes,
+                        activation=jax.nn.relu,
+                        activate_final=activate_mlp_final,
+                        w_init=hk.initializers.Orthogonal(scale=jnp.sqrt(2)),
```

Comment:
We can rather create two lambda functions depending on `if orthogonal_initialisation`. And then just call them inside the policy_fn and critic_fn. Instead of duplicating the policy and critic code.

Reply:
I like this a lot more! Will update :)

Reply:
Done. Please let me know your thoughts. :)